src/libraries/System.Drawing.Common/src/System/Drawing/Image.cs

```diff
@@ -298,6 +298,65 @@ public PixelFormat PixelFormat
             }
         }
 
+        /// <summary>
+        /// Gets an array of the property IDs stored in this <see cref='Image'/>.
+        /// </summary>
+        [Browsable(false)]
+        public unsafe int[] PropertyIdList
+        {
+            get
+            {
+                Gdip.CheckStatus(Gdip.GdipGetPropertyCount(new HandleRef(this, nativeImage), out uint count));
+                if (count == 0)
+                    return Array.Empty<int>();
+
+                var propid = new int[count];
+                fixed (int* pPropid = propid)
+                {
+                    Gdip.CheckStatus(Gdip.GdipGetPropertyIdList(new HandleRef(this, nativeImage), count, pPropid));
+                }
+
+                return propid;
+            }
+        }
+
+        /// <summary>
+        /// Gets an array of <see cref='PropertyItem'/> objects that describe this <see cref='Image'/>.
+        /// </summary>
+        [Browsable(false)]
+        public unsafe PropertyItem[] PropertyItems
+        {
+            get
+            {
+                Gdip.CheckStatus(Gdip.GdipGetPropertySize(new HandleRef(this, nativeImage), out uint size, out uint count));
+
+                if (size == 0 || count == 0)
+                    return Array.Empty<PropertyItem>();
+
+                PropertyItemInternal* propdata = (PropertyItemInternal*)Marshal.AllocHGlobal((int)size);
```

Comment:
Can you change this to rent a byte array from the ArrayPool and pin that for the buffer please? It would be good to get rid of the AllocHGlobal. :)